### PR TITLE
[WEB-3772] logging environments

### PR DESCRIPTION
**Pull request by tadam313**
* differentiate local and prod logging
* additional tests
* expose: selected menu, app slug in web mode and mode to global context

### CODE REVIEW

**Comment by tadam313** [comment]
@simonmarton need to merge your logger branch first then conflicts then this :)

File: source/javascripts/services/logger.ts [hunk 1]
```diff
@@ -1,41 +1,89 @@
-import { datadogLogs } from "@datadog/browser-logs";
+import { pick } from "underscore";
+import { datadogLogs, StatusType, HandlerType, Logger as DLogger } from "@datadog/browser-logs";
 import { Context } from "@datadog/browser-core";
+import { getAppSlug } from "./app-service";
 import { WFEWindow } from "../typings/global";
 
+const w = window as WFEWindow;
```
Comment (simonmarton) [inline source/javascripts/services/logger.ts]: Good idea, we can re-declare `window` type
```suggestion
declare var window: WFEWindow;
```

File: spec/javascripts/logger.spec.js [hunk 2]
```diff
@@ -1,54 +1,75 @@
...
+		expect(mockInnerLogger.debug).toHaveBeenCalledWith("test", { test: "ctx" });
 	});
 
-	it("should use datadog info logging", () => {
-		logger.info("test", { test: "ctx" });
-		expect(mockDatadogLogs.logger.info).toHaveBeenCalledWith("test", { test: "ctx" });
-	});
+  it("should use datadog info logging", () => {
+    logger.info("test", "test-ctx");
+    expect(mockInnerLogger.info).toHaveBeenCalledWith("test", "test-ctx");
```
Comment (simonmarton) [inline spec/javascripts/logger.spec.js]: Please test for a context object

File: bitrise.yml [hunk 3]
```diff
@@ -90,6 +90,7 @@ workflows:
     envs:
       - NODE_ENV: "prod"
       - MODE: "WEBSITE"
+      - ANALYTICS: true
```
Comment (tadam313) [inline bitrise.yml]: idea: to be more flexible introduces analytics property

File: source/javascripts/_BitriseWorkflowEditor.js.erb [hunk 4]
```diff
@@ -1,4 +1,7 @@
-window.DATADOG_API_KEY = "<%= data[:constants][:datadog][:apiKey] %>";
+window.serviceName = "workflow-editor";
+window.datadogApiKey = "<%= data[:constants][:datadog][:apiKey] %>";
+window.isAnalyticsOn = <%= analytics? %>;
```
Comment (tadam313) [inline source/javascripts/_BitriseWorkflowEditor.js.erb]: based on env var and build environment

**Pull request merged**